File: src/jzrlib/external/markovchains/CharChain.java
package jzrlib.external.markovchains;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Serg V. Zhdanovskih
 */
public final class CharChain extends MarkovChain<Character>
{
    public CharChain(int order)
    {
        super(order, ' ');
    }

    public final void init(String data)
    {
        List<Character> chars = new ArrayList<>();
        for (char c : data.toCharArray()) {
            chars.add(c);
        }

        this.addItems(chars);
    }
}

File: project/External/MarkovChains/CharChain.cs
using System;
using System.Collections.Generic;

namespace ZRLib.External.MarkovChains
{
    public sealed class CharChain : MarkovChain<Char>
    {
        public CharChain(int order)
            : base(order, ' ')
        {
        }

        public void Init(string data)
        {
            IList<Char> chars = new List<Char>();
            foreach (char c in data) {
                chars.Add(c);
            }

            AddItems(chars);
        }
    }
}
